src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -132,516 +116,199 @@ MasterTargetList(List *workerTargetList)
-	 * on distinct clauses. Therefore we can safely append distinct clauses to
-	 * the end of order by clauses. Although the same column may appear more
-	 * than once in order by clauses, created plan uses only one instance, for
-	 * example order by a,b,a,a,b,c is translated to equivalent order by a,b,c.
+	 * 100k on the row estimate causes the postgres planner to behave very much like the
+	 * old citus planner in the plans it produces. Namely the old planner had hardcoded
+	 * the use of Hash Aggregates for most of the operations, unless a postgres guc was
+	 * set that would disallow hash aggregates to be used.
 	 *
-	 * If the query has distinct clause but not distinct on, we first create
-	 * distinct plan that is either HashAggreate or Sort + Unique plans depending
-	 * on hashable property of columns in distinct clause. If there is order by
-	 * clause, it is handled after distinct planning.
-	 */
-	if (masterQuery->hasDistinctOn)
-	{
-		ListCell *distinctCell = NULL;
-		foreach(distinctCell, masterQuery->distinctClause)
-		{
-			SortGroupClause *singleDistinctClause = lfirst(distinctCell);
-			Index sortGroupRef = singleDistinctClause->tleSortGroupRef;
-
-			if (get_sortgroupref_clause_noerr(sortGroupRef, sortClauseList) == NULL)
-			{
-				sortClauseList = lappend(sortClauseList, singleDistinctClause);
-			}
-		}
-	}
-	else if (masterQuery->distinctClause)
-	{
-		Plan *distinctPlan = BuildDistinctPlan(masterQuery, topLevelPlan);
-		topLevelPlan = distinctPlan;
-	}
-
-	/* (4) add a sorting plan if needed */
-	if (sortClauseList)
-	{
-		Sort *sortPlan = make_sort_from_sortclauses(sortClauseList, topLevelPlan);
-
-		/* just for reproducible costs between different PostgreSQL versions */
-		sortPlan->plan.startup_cost = 0;
-		sortPlan->plan.total_cost = 0;
-		sortPlan->plan.plan_rows = 0;
-
-		topLevelPlan = (Plan *) sortPlan;
-	}
-
-	/*
-	 * (5) add a unique plan for distinctOn.
-	 * If the query has distinct on we add a sort clause in step 3. Therefore
-	 * Step 4 always creates a sort plan.
-	 * */
-	if (masterQuery->hasDistinctOn)
-	{
-		Assert(IsA(topLevelPlan, Sort));
-		topLevelPlan =
-			(Plan *) make_unique_from_sortclauses(topLevelPlan,
-												  masterQuery->distinctClause);
-	}
-
-	/* (5) add a limit plan if needed */
-	if (masterQuery->limitCount || masterQuery->limitOffset)
-	{
-		Node *limitCount = masterQuery->limitCount;
-		Node *limitOffset = masterQuery->limitOffset;
-		Limit *limitPlan = make_limit(topLevelPlan, limitOffset, limitCount);
-		topLevelPlan = (Plan *) limitPlan;
-	}
-
-	/*
-	 * (6) set top level plan in the plantree and copy over some things from
-	 * PlannerInfo
+	 * Ideally we would be able to provide estimates close to postgres' estimates on the
+	 * workers to let the standard planner choose an optimal solution for the masterQuery.
 	 */
-	FinalizeStatement(root, selectStatement, topLevelPlan);
+	path->custom_path.path.rows = 100000;
```

Comment:
not super important, but it's a little odd that we set rows without setting cost.